### PR TITLE
Drop support of < PHP 7.2

### PR DESCRIPTION
**Pull request by odolbeau**
+ Raise phpunit to 8.4
+ Improve travis test suite to test against all sf versions.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,5 +3,5 @@
 DIR := ${CURDIR}
 QA_IMAGE := jakzal/phpqa:php7.3-alpine
 
-static:
+cs-fix:
```
Comment (odolbeau) [inline Makefile]: @stof Fixed

File: .travis.yml [hunk 2]
```diff
@@ -16,20 +16,32 @@ env:
...
+
+        # Test with latest PHP version & beta dependencies
+        - { php: 7.4snapshot, env: STABILITY=dev }
+
+    allow_failures:
+        - { php: 7.4snapshot, env: STABILITY=dev }
 
 before_install:
-  - if [ "$DEPENDENCIES" == "beta" ]; then composer config minimum-stability beta; fi;
+  - if [ "$STABILITY" == "beta" ]; then composer config minimum-stability beta; fi;
```
Comment (stof) [inline .travis.yml]: isn't this already handled by the next line ?

File: .travis.yml [hunk 3]
```diff
@@ -16,20 +16,32 @@ env:
         - PHPAMQP_VERSION=v1.9.4
 
 matrix:
+    fast_finish: true
     include:
-        # Test with default configuration
-        - { php: 7.2 }
-        - { php: 7.3 }
+        # Test with lowest dependencies
+        - { php: 7.2, env: COMPOSER_FLAGS='--prefer-lowest --prefer-stable' }
```
Comment (stof) [inline .travis.yml]: I would keep a PHP 7.2 test using normal dependencies.

File: .travis.yml [hunk 4]
```diff
@@ -16,20 +16,32 @@ env:
...
 
-    fast_finish: true
+        # Force some major versions of Symfony
+        - php: 7.3
+          env: SYMFONY_VERSION="3.*"
+        - php: 7.3
+          env: SYMFONY_VERSION="4.*"
+
+        # Test with latest PHP version & beta dependencies
+        - { php: 7.4snapshot, env: STABILITY=dev }
```
Comment (stof) [inline .travis.yml]: should be `7.4` now rather than `7.4snapshot`

**Pull request merged**